packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1379,10 +1379,15 @@ class Navigator extends StatefulWidget {
   /// This callback is responsible for calling [Route.didPop] and returning
   /// whether this pop is successful.
   ///
-  /// The [Navigator] widget should be rebuilt with a [pages] list that does not
-  /// contain the [Page] for the given [Route]. The next time the [pages] list
-  /// is updated, if the [Page] corresponding to this [Route] is still present,
-  /// it will be interpreted as a new route to display.
+  /// This callback is also responsible for updating the list of [Page]s that
```

Comment:
The old doc was a little clearer because it explained what "updating the list" means (i.e. remove the page of this route from the list). Why make this weaker?

Reply:
What if user does want to have the page persist in the list? I feel kinda weird to ask user to remove the page, so i update the document to be more generic and state what will happen if the page is still persist in the list.

Reply:
In all use cases I can think of you would remove the page from the list after you decided to pop it. We should be clear in the docs that that's the expected behavior and then point out what happens if you don't do that.